src/framework/control.ml

```diff
@@ -304,11 +304,11 @@ struct
 
     GU.global_initialization := true;
     GU.earlyglobs := get_bool "exp.earlyglobs";
-    let marshal =
+    let marshal: Spec.marshal option =
       if get_string "load_run" <> "" then
         Some (Serialize.unmarshal Fpath.(v (get_string "load_run") / "spec_marshal"))
       else if Serialize.results_exist () && get_bool "incremental.load" then
-        Some (Serialize.load_data Serialize.AnalysisData)
+        Some (Obj.obj (Serialize.Cache.(get_data AnalysisDataRequest)))
```

Comment:
Might be better to move the `Obj.obj` inside `get_data`, so from the outside, it is all type-safe. At least previously `Control` didn't have to do those dangerous conversions itself.

Reply:
Maybe if instead of `| SolverData : Obj.t data_query` it is defined as `| SolverData : _ data_query` (that particular query being of any type, it might work. Of course it cannot have wildcard type in the record, so it'll have to stay `Obj.t` there, but the `Obj.obj`/`Obj.repr` could be used in the corresponding cases in `get_data` and `update_data`.

Reply:
Good idea, that seems to work!